File: project/AccessControlSystem-Server/src/main/java/com/acs/model/Action.java
package com.acs.model;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name="acs_action")
public class Action {

    @Id
    @Column(name="id")
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private int id;

    @ManyToOne(targetEntity = Employee.class)
    private Employee employee;

    private Direction direction;

    private Date gendate;

    @OneToOne(targetEntity = OfficeRoom.class)
    private OfficeRoom officeRoom;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Employee getEmployee() {
        return employee;
    }

    public void setEmployee(Employee employee) {
        this.employee = employee;
    }

    public Direction getDirection() {
        return direction;
    }

    public void setDirection(Direction direction) {
        this.direction = direction;
    }

    public Date getGendate() {
        return gendate;
    }

    public void setGendate(Date gendate) {
        this.gendate = gendate;
    }

    public OfficeRoom getOfficeRoom() {
        return officeRoom;
    }

    public void setOfficeRoom(OfficeRoom officeRoom) {
        this.officeRoom = officeRoom;
    }
}

File: project/AccessControlSystem-Client/src/app/components/models/Action.ts
export class Action {

  id: string;
  employee: string;
  direction: string;
  officeRoom: string;
  doorLock: string[];
  gendate: string;

  constructor(id: string, employee: string, direction: string, officeRoom: string, doorLock: string[], gendate: string) {
    this.id = id;
    this.employee = employee;
    this.direction = direction;
    this.officeRoom = officeRoom;
    this.doorLock = doorLock;
    this.gendate = gendate;
  }
}
